examples/list_box_model/model/mod.rs

```diff
@@ -0,0 +1,41 @@
+//! Defines our custom model
+
+mod imp;
+
+use crate::row_data::RowData;
+use glib::subclass::prelude::*;
+use gtk::{gio, glib, prelude::*};
+
+// Public part of the Model type.
+glib::wrapper! {
+    pub struct Model(ObjectSubclass<imp::Model>) @implements gio::ListModel;
+}
+
+// Constructor for new instances. This simply calls glib::Object::new()
+impl Model {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Model {
+        glib::Object::new(&[]).expect("Failed to create Model")
+    }
+
+    pub fn append(&self, obj: &RowData) {
+        let self_ = imp::Model::from_instance(self);
+        let index = {
+            // Borrow the data only once and ensure the borrow guard is dropped
+            // before we emit the items_changed signal because the view
+            // could call get_item / get_n_item from the signal handler to update its state
+            let mut data = self_.0.borrow_mut();
+            data.push(obj.clone());
+            data.len() - 1
+        };
+        // Emits a signal that 1 item was added, 0 removed at the position index
+        self.items_changed(index as u32, 0, 1);
+    }
+
+    pub fn remove(&self, index: u32) {
+        let self_ = imp::Model::from_instance(self);
```

Comment:
Same here, ideally we would do that for all the examples but at least newer ones should use it